### ISSUE TITLE
API changes for trajectory cost functions

### ISSUE THREAD

**Issue by PeterMitrano** [DLu/additional_dwa_plugins#1]
I realize this repo is very old and not maintained, but I'm curious if these could be adapted to use with indigo. The functions `initialize` and `setGlobalPlan` don't seem to be part of the `base_local_planner::TrajectoryCostFunction` class (anymore? where they in the past?)

What would the modern approach be if I want to ensure my robot turns towards it's path before moving forward? (not that indigo is modern)

**Comment by DLu** [comment]
Hi @PeterMitrano . What do you mean adapted to use with inidigo? Does it not compile?

**Comment by PeterMitrano** [comment]
Correct. After updating to use catkin, I see that

```
fatal error: dwa_local_planner/trajectory_cost_function.h: No such file or directory
```

Turns out that file has moved from `dwa_local_planner` to `base_local_planner`.

There are a ton of other fairly trivial errors that I could fix. But the bigger issue is that there are functions you use, `initialize` and `setGlobalPlan` that don't seem to be part of the new `TrajectoryCostFunction` interface anymore. Because of the lack of any initialize method, I'm not sure it's possible to have a trajectory cost that subscribes to  things like robot footprint before scoring functions.

I guess I don't have any real question, and I've already switch to eband so I'm closing this.

But if you have a second, I've made a PR to move_base that no one has looked at yet:
https://github.com/ros-planning/navigation/pull/502
